app/controllers/events_controller.rb

```diff
@@ -0,0 +1,112 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+  before_action :set_current_user, only: %i[show index new attend_events show_events]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
+    @upcoming_events = Event.upcoming_events
+    @previous_events = Event.previous_events
+  end
+
+  # GET /events/1
+  # GET /events/1.json
+  def show
+    @creator = creator
+  end
+
+  # GET /events/new
+  def new
+    @event = Event.new
+  end
+
+  # GET /events/1/edit
+  def edit; end
+
+  # POST /events
+  # POST /events.json
+  def create
+    @event = current_user.events_created.build(event_params)
+
+    respond_to do |format|
+      if @event.save
+        format.html { redirect_to @event, notice: 'Event was successfully created.' }
+        format.json { render :show, status: :created, location: @event }
+      else
+        format.html { render :new }
+        format.json { render json: @event.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /events/1
+  # PATCH/PUT /events/1.json
+  def update
+    respond_to do |format|
+      if @event.update(event_params)
+        format.html { redirect_to @event, notice: 'Event was successfully updated.' }
+        format.json { render :show, status: :ok, location: @event }
+      else
+        format.html { render :edit }
+        format.json { render json: @event.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /events/1
+  # DELETE /events/1.json
+  def destroy
+    @event.destroy
+    respond_to do |format|
+      format.html { redirect_to events_url, notice: 'Event was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  def show_events
+    @events = Event.all.reject { |event| @current_user.attended_events.include?(event) }
+  end
+
+  def attend_events
```

Comment:
Custom methods like these shouls be on the event model instead of the controllers.
You can create a scope for `show_events` and a method for `attend_events`.

Remember the rule "thin controllers, thick models"